### PR TITLE
Add done() API to Future

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42013 Add done() API to Future**

Differential Revision: [D22729596](https://our.internmc.facebook.com/intern/diff/D22729596)

### CODE REVIEW

**Comment by mrshenli** [comment]
Screen Shot 2020-07-24 at 12 00 23 PM

File: torch/_C/__init__.pyi.in [hunk 1]
```diff
@@ -120,6 +120,7 @@ class _LegacyVariableBase(object):
 # Defined in torch/csrc/jit/python/init.cpp
 class Future(object):
   def __init__(self) -> None: ...
+  def done(self) -> _bool: ...
```
Comment (mrshenli) [inline torch/_C/__init__.pyi.in:123]: @malfet is this the right way to annotate?

**Review by rohan-varma — APPROVED** [review]
LGTM! For the annotation, is there any way to test/validate it (such as running mypy typechecker)?

**Comment by mrshenli** [comment]
> LGTM! For the annotation, is there any way to test/validate it (such as running mypy typechecker)?

I believe there are tests to cover this. The first version caused test failures, as I was using `bool` instead of `_bool`.

**Review by pbelevich — APPROVED** [review]

File: torch/_C/__init__.pyi.in (hunk 1 above)
Comment (malfet) [reply torch/_C/__init__.pyi.in:123]: Yes, thank you for adding the annotation.

**Pull request closed without merging**